esp-hal/src/uart.rs

```diff
@@ -915,47 +916,78 @@ where
         // initialize peripheral by setting clk_enable and clearing uart_reset bits
         T::enable_peripheral();
 
+        // don't reset the console UART - this will cause trouble at least on ESP32-S3
```

Comment:
Mainly for my context, would it be possible to explain when it _would_ be okay to reset the uart being as the console? For instance, if we make it configurable later, why would someone want to enable that config option later?